lib/data/replacements.json

```diff
@@ -669,6 +670,18 @@
       }
     ]
   },
+  "gradle-wrapper-validation-action": {
+    "description": "gradle/wrapper-validation-action got renamed to gradle/actions/wrapper-validation",
+    "packageRules": [
+      {
+        "matchCurrentVersion": "3",
```

Comment:
Are you sure this rule is correct? It looks like the current rule will only match on precisely version `3`?

What if the user has a version within the `3` version range, like:

- `3.0.0`
- `3.1.0`
- `3.1.1`
- `3.9.9`

Reply:
I see, so the rule is incorrect. How should I fix it? Replace it by a regex that matches `3` and `3.x.y`? But what to use as `replacementVersion` then or can I remove it?

Reply:
## How to improve your preset

> I see, so the rule is incorrect. How should I fix it? Replace it by a regex that matches `3` and `3.x.y`? But what to use as `replacementVersion` then or can I remove it?

What I think you should do:

- The `matchCurrentVersion` on the _old_ package should be a regex that matches `3` and `3.x.y.`. 
- The `replacementVersion` should be the _first_ recommended version of the _new_ dependency. Pin the new version, by setting a specific new version.

## Source for changes

My changes are based on the rules listed in the [Renovate developer docs, Creating/editing Renovate presets](https://github.com/renovatebot/renovate/blob/main/docs/development/creating-editing-renovate-presets.md). Full quote of the section for replacement presets:

> #### Rules for replacement presets
> 
> 1. Replacement PRs should ideally propose a replacement only once the user is on a compatible version, by specifying a compatible `matchCurrentVersion` constraint
> 1. If no compatible replacement upgrade is possible, it's acceptable to propose an incompatible one (e.g. a major version upgrade)
> 1. Replacements should update the user to the first recommended version of the new dependency and not include any new changes - whether breaking or not - if they can be avoided, in short: pin the new version
> 
> If possible, replacement presets should give the user a replacement version that is _functionally identical_ to the _last version_ under the _old_ name.
> We only want a user's tests to fail because of _broken references_ to the old package name, and not because the maintainer(s) changed the _behavior_ of the package.

Reply:
Ok, I updated both values.